src/xenia/base/math.h

```diff
@@ -59,6 +60,36 @@ T next_pow2(T value) {
   return value;
 }
 
+#if __cplusplus >= 201703L
+template <typename T>
+inline constexpr T greatest_common_divisor(T a, T b) {
+  return std::gcd(a, b);
+}
+#else
+template <typename T>
+constexpr T greatest_common_divisor(T a, T b) {
+  // Use the Euclid algorithm to calculate the greatest common divisor
+  while (b) {
+    T tmp = b;
```

Comment:
```suggestion
    T tmp = std::exchange(b, a % b);
```

This will do nothing to codegen, but I find it much nicer looking than splitting such idioms in two lines like it's done currently. Up to you whether you want to go for it though, as it's just a styling change.

Reply:
Then
```
while(b)
{
  a = std::exchange(b, a % b);
}
```
I guess?

Reply:
Indeed! That will work, since `std::exchange` returns by value and not a reference.